src/app/[entry_id]/page.jsx

```diff
@@ -0,0 +1,100 @@
+"use client"
+
+import React from 'react'
+import { useEffect, useState } from 'react'
+import { supabase } from '../lib/supabase'
+import Link from 'next/link'
+import Image from 'next/image'
+
+function Entry({ params }) {
+  const [entryData, setEntryData] = useState();
+  const entry_id = params.entry_id
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        let { data: diary_entry, error } = await supabase
+          .from('diary_entries')
+          .select()
+          .eq('id', entry_id)
+        if (error) {
+          throw error;
+        }
+        setEntryData(diary_entry[0])
+      } catch (error) {
+        console.error('Error fetching data from Supabase:', error.message);
+      }
+    };
+    fetchData();
+  }, []);
```

Comment:
Would look at potentially creating a `useSiteDiaries` hook to group all the queries and mutations into a common place. Not a biggie though.

Possibly what is even better is to do this query server side in NextJs rather than in a `useEffect`. My life's mission is to remove as many `useEffect`s as possible 😂